internal/scanner/scanner.go

```diff
@@ -119,6 +119,32 @@ func (s *Scanner) CheckGRPCAvailability(ctx context.Context) {
 	s.db.IsGrpcAvailable = available
 }
 
+// CheckGraphQlAvailability checks if the GraphQL is available at the given URL.
+func (s *Scanner) CheckGraphQlAvailability(ctx context.Context) {
+	s.logger.WithField("status", "started").Info("GraphQL pre-check")
+
+	available, err := s.checkGraphQlAvailability(ctx)
+	if err != nil {
+		s.logger.WithFields(logrus.Fields{
+			"status":     "done",
+			"connection": "not available",
+		}).WithError(err).Infof("GraphQL pre-check")
+	}
+	if available {
+		s.logger.WithFields(logrus.Fields{
+			"status":     "done",
+			"connection": "available",
+		}).Info("GraphQL pre-check")
+	} else {
```

Comment:
Can't we simplify this tree?
Something like:
1. if err != nil or not available = "not available"
2. else (... if err is nil and it is available) = "available"

Reply:
These two blocks (error or not available) are the same basically

Reply:
Fixed - https://github.com/wallarm/gotestwaf/pull/167/commits/30ea977db6ab0bd30606474a186f16a33b012a9f